### ISSUE TITLE
Additional relationship support: hasOneOfDescendantsAndSelf

### ISSUE THREAD

**Issue by mrgodhani** [staudenmeir/laravel-adjacency-list#60]
Hi there,

Is there any support or idea how to implement hasOne for descendants.  Trying to eager load descendants to include hasOne as well.

Also when I try `eloquent-param-limit-fix` to solve param limit issue in MSSQL it doesn't quite work with descendants.

**Comment by staudenmeir** [comment]
I'll look into it.

**Comment by staudenmeir** [comment]
Can you share your use case(s)?

> Also when I try eloquent-param-limit-fix to solve param limit issue in MSSQL it doesn't quite work with descendants.

Are you referring to https://github.com/staudenmeir/laravel-adjacency-list/issues/6#issuecomment-850840278 or is this a different issue?

**Comment by mrgodhani** [comment]
@staudenmeir 

For the `eloquent-param-limit-fix` yes I was referring to #6. 

Here is the use case we have in our product.

```
class Ruleset {
use HasRecursiveRelationships;

public function behavior()
    {
        return $this
            ->hasOne(Behavior::class, 'id', 'behavior_id')
            ->withTrashed();
    }
}
```

My use case is to include behavior and making the query easier with that relation vs at currently what we are doing is following.

```
class Flow {

    public function rulesets()
    {
        return $this->hasMany(Ruleset::class, 'workflow_id', 'id')
            ->with('descendants')
            ->isRoot();
    }
}
```

I wonder above can be changed to  something like below instead

```
    public function rulesets()
    {
        return $this->hasMany(Ruleset::class, 'flow_id', 'id')
            ->with('recursiveBehaviors');
    }
```

If it makes sense.  Looking at the example in the readme that's what I making the assumption.
